src/components/application_manager/src/application_manager_impl.cc

```diff
@@ -2068,6 +2068,38 @@ void ApplicationManagerImpl::ProcessServiceStatusUpdate(
       service_type, service_event, reason, app_id);
 
   rpc_service_->ManageHMICommand(notification);
+
+  if (hmi_apis::Common_ServiceEvent::REQUEST_REJECTED == service_event) {
+    StopServicesForApp(app, service_type);
+    state_ctrl_.SetRegularState(app,
+                                mobile_apis::PredefinedWindows::DEFAULT_WINDOW,
+                                mobile_apis::HMILevel::HMI_NONE,
+                                true);
+  }
+}
+
+void ApplicationManagerImpl::StopServicesForApp(
```

Comment:
What is the purpose of this method? I dont really follow how this addresses the original issue? Why is the only case for VIDEO and not audio/rpc?

Also 

> navi_service_status_.erase(app->app_id());

isnt really stopping anything, its just deleting the app id from a data structure.